File: Source/WickedSick.Server.XamlParser/Elements/Media/Animation/BeginStoryboard.cs

namespace WickedSick.Server.XamlParser.Elements.Media.Animation
{
    [Element("Fayde.Media.Animation")]
    public class BeginStoryboard : TriggerAction
    {
        public static readonly PropertyDescription StoryboardProperty = PropertyDescription.Register("Storyboard", typeof(Storyboard), typeof(BeginStoryboard), true);
    }
}

File: Source/WickedSick.Fayde.Client.Engine/Javascript/Media/Animation/BeginStoryboard.js
/// <reference path="../../Core/Triggers.js"/>
/// <reference path="Storyboard.js"/>
/// CODE

(function (namespace) {
    var BeginStoryboard = Nullstone.Create("BeginStoryboard", Fayde.TriggerAction);

    //#region Properties

    BeginStoryboard.StoryboardProperty = DependencyProperty.Register("Storyboard", function () { return namespace.Storyboard; }, BeginStoryboard);

    Nullstone.AutoProperties(BeginStoryboard, [
        BeginStoryboard.StoryboardProperty
    ]);

    //#endregion

    //#region Annotations

    BeginStoryboard.Annotations = {
        ContentProperty: BeginStoryboard.StoryboardProperty
    };

    //#endregion

    BeginStoryboard.Instance.Fire = function () {
        var sb = this.Storyboard;
        if (sb)
            sb.Begin();
    };

    namespace.BeginStoryboard = Nullstone.FinishCreate(BeginStoryboard);
})(Nullstone.Namespace("Fayde.Media.Animation"));
